### ISSUE TITLE
SVG Interactivity

### ISSUE THREAD

**Issue by NoSuck** [qutebrowser/qutebrowser#3485]
SVG allows for [interactivity through pointer events.](https://www.w3.org/TR/SVG11/interact.html#SVGEvents)  It would be nice if SVG elements specified as interactive by `<set>` elements (e.g. `circleBlue` in `circleBlue.mousedown` of the attached demonstration) could have hintchars.

Thank you for the greatest web browser I have ever had the privilege of using.

[qbSvgInteractivityDemo.zip](https://github.com/qutebrowser/qutebrowser/files/1628217/qbSvgInteractivityDemo.zip)

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="6cm" height="5cm" viewBox="0 0 600 500" xmlns="http://www.w3.org/2000/svg" version="1.1">
	<desc>SVG interactivity demo for qutebrowser</desc>

	<!-- border -->
	<rect x="1" y="1" width="598" height="498" fill="none" stroke="black"/>

	<!-- https://www.w3.org/TR/SVG11/interact.html#SVGEvents -->
	<circle id="circleRed" cx="150" cy="225" r="100" fill="red">
		<set attributeName="fill" to="blue" begin="circleBlue.mousedown"/>
	</circle>

	<circle id="circleBlue" cx="450" cy="225" r="100" fill="blue"/>

	<text x="300" y="480" font-size="28" text-anchor="middle">
		Click blue circle to change red circle.
	</text>
</svg>
```

**Comment by The-Compiler** [comment]
I need to think about this some more, but I think I'm leaning towards "not worth the complexity" here - can we get the position of such elements via JS somehow or would we need to have some kind of SVG parser?

**Comment by NoSuck** [comment]
You're right about the work/benefit ratio.  Most web developers have yet to realize the power of SVG, and proliferation (particularly with concern to interactivity) fails to reflect the format's soundness of design.  I hope a lack of popularity isn't grounds for closing this issue.

[DOM Element.getBoundingClientRect()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) has been [cited](https://stackoverflow.com/a/19203552/2230956) as the easiest way to retrieve a given SVG element's location.  The problem then would be determining which elements contain [interactive events.](https://www.w3.org/TR/SVG11/interact.html#SVGEvents)
